File: server/parse-pdf.js
const pdf = require('pdf-parse/lib/pdf-parse.js');
const fs = require('fs');
const buf = fs.readFileSync('../DOC/CONTRATO.pdf');
pdf(buf).then(d => {
    console.log('PAGES:', d.numpages);
    console.log('--- TEXT ---');
    console.log(d.text);
}).catch(e => console.error('ERROR:', e.message));

File: server/parse-pdf.py
from PyPDF2 import PdfReader

r = PdfReader(r'c:\Projeto Code IA\fncd-capital\DOC\CONTRATO.pdf')
with open('pdf-text.txt', 'w', encoding='utf-8') as f:
    f.write(f'Total Pages: {len(r.pages)}\n')
    for i, p in enumerate(r.pages):
        f.write(f'\n\n========== PAGE {i+1} ==========\n\n')
        text = p.extract_text()
        if text:
            f.write(text)
        else:
            f.write('[no text extracted]')
print('Done! Written to pdf-text.txt')
